activerecord/lib/active_record/migration.rb

```diff
@@ -1395,11 +1402,14 @@ def validate(migrations)
       end
 
       def record_version_state_after_migrating(version)
-        if down?
+        if ENV["DRY_RUN_MIGRATION"] = "true"
+          # do nothing
+        elsif down?
           migrated.delete(version)
           @schema_migration.delete_by(version: version.to_s)
         else
           migrated << version
+          # How does this have a connection that is actually executed?
```

Comment:
I should look at this object and see about its inheritance. Maybe what it depends on for the connection is an instance that exists before DryConnection is prepended?